wgpu-core/src/command/bundle.rs

```diff
@@ -1036,14 +1036,30 @@ pub mod bundle_ffi {
         stages: wgt::ShaderStage,
         offset: u32,
         size_bytes: u32,
-        data: *const u32,
+        data: *const u8,
     ) {
         span!(_guard, DEBUG, "RenderBundle::set_push_constants");
-        let data_slice = slice::from_raw_parts(data, (size_bytes / 4) as usize);
+        assert_eq!(
+            offset & 3,
+            0,
+            "Push constant offset must be aligned to 4 bytes."
+        );
+        assert_eq!(
+            size_bytes & 3,
+            0,
+            "Push constant size must be aligned to 4 bytes."
+        );
+        let data_slice = slice::from_raw_parts(data, size_bytes as usize);
         let value_offset = pass.base.push_constant_data.len().try_into().expect(
             "Ran out of push constant space. Don't set 4gb of push constants per RenderBundle.",
         );
-        pass.base.push_constant_data.extend_from_slice(data_slice);
+
+        pass.base.push_constant_data.extend(
+            data_slice
+                .chunks_exact(4)
```

Comment:
why not just `chunk`? we already checked the size to be aligned to 4

Reply:
This was my doing in the code I suggested, iirc the code generated for chunk_exact was better than chunk as it can statically guarantee the accesses won't panic.

Reply:
See https://godbolt.org/z/Wf619s